scripts/settings_list.py

```diff
@@ -0,0 +1,53 @@
+import yaml
+import pprint
+
+class SettingsList():
+  list_of_dicts= list();
+
+  def sl_print(self):
+    pp = pprint.PrettyPrinter(indent=2)
+    pp.pprint(self.list_of_dicts)
+
+  def sl_get_dict(self, group, name):
+    for element in self.list_of_dicts:
+        if element['name'] == name and element['group'] == group:
+          return element
+
+  def sl_get_field(self, group, name,field):
+    thisdict = self.sl_get_dict(group, name)
+    returnvar = ""
+    if thisdict and isinstance(thisdict, dict):
+      returnvar = thisdict.get(field,"")
+    else:
+      print "Error in settings list parsed yaml file."
+      print "No dictionary dictionary for name  {0} and group is {1}".format(name,group)
+    if not returnvar:
+      returnvar = ""
+    return returnvar
+
+  def sl_return_groups(self):
+    group_set = set()
+    output = []
+    for element in self.list_of_dicts:
+        group = element['group']
+        if group not in group_set:
+          output.append(element['group'])
+          group_set.add(group)
+    return output
+
+
+  def __init__(self,path_to_file):
+    stram = open(path_to_file, "r")
+    self.list_of_dicts = yaml.load(stram)
+    # Debug
+    # print "Loaded settings yamle file from path " + path_to_file
+    # print "Number of settings loaded {0}".format(len(self.list_of_dicts))
+    # print "Type of the first element in the file is {0}".format(type(self.list_of_dicts[0]))
+
+
+# Test code
```

Comment:
You could turn this into a simple unit test, Buro has been using `py.test` with the sbp_log_analysis stuff and its really easy to use, you can just make functions called `test_xxx` and it does the rest. Doesn't get in the way of the code.

If you do this we can set up jenkins to check the tests as part of the build process.